ui/e2e/explorer/workflow/instance.spec.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  createInstance,
+  simpleWorkflow as simpleWorkflowContent,
+} from "./utils";
+import { createNamespace, deleteNamespace } from "../../utils/namespace";
+import { expect, test } from "@playwright/test";
+
+import { createFile } from "e2e/utils/files";
+import { faker } from "@faker-js/faker";
+
+let namespace = "";
+const simpleWorkflowName = faker.system.commonFileName("yaml");
+
+test.beforeEach(async () => {
+  namespace = await createNamespace();
+  /* create workflows we can use to create instances later */
+  await createFile({
+    name: simpleWorkflowName,
+    namespace,
+    type: "workflow",
+    yaml: simpleWorkflowContent,
+  });
+});
+
+test.afterEach(async () => {
+  await deleteNamespace(namespace);
+  namespace = "";
+});
+
+test("the header of the instance page shows the relevant data for the workflow", async ({
+  page,
+}) => {
+  const newInstance = createInstance({ namespace, path: simpleWorkflowName });
+
+  await expect(newInstance, "wait until process was completed").toBeDefined();
+
+  const instanceID = (await newInstance).instance;
+
+  await page.goto(`/${namespace}/instances/${instanceID}`);
+
+  const header = page.getByTestId("instance-header-container");
+  await expect(header, "the header is visible").toBeVisible();
+
+  const instanceID_Header = header.locator("h3");
+  await expect(instanceID_Header, "the instance ID is visible").toBeVisible();
```

Comment:
This test could be a bit more specific. The description says "the instance ID is visible" but it actually only checks if the header has a h3 tag.

And a small suggestion about test descriptions: I'm not very good with test descriptions myself, but from time to time I spot a description that might need a little tweaking :D So "the instance ID is visible" might just be "It renders the instance ID". Just a small detail, but "is visible" somehow implies to me, that this ID is not visible all the time, like e.g. a tooltip. 

[Sebastian is very good at this stuff](https://github.com/direktiv/wiki/wiki/Dev-(UI):-Writing-e2e-tests-with-playwright#best-practices)